src/main/java/com/minecolonies/coremod/colony/managers/interfaces/IRaiderManager.java

```diff
@@ -0,0 +1,119 @@
+package com.minecolonies.coremod.colony.managers.interfaces;
+
+import com.minecolonies.coremod.entity.ai.mobs.AbstractEntityMinecoloniesMob;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import net.minecraft.world.WorldServer;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.List;
+
+/**
+ * Interface implementing all methods required for all raider managers.
+ */
+public interface IRaiderManager
```

Comment:
So I like the change to the raider terminology. But I think we can improve this.
Right now we have special handling for ships. Is their a reason why we can not make this more generic? EG Support barbarian camps spawning and despawning for barbarians.

Maybe we may want to add support for a third raider type, like vikings when you are in a cold biome, or something special for desserts?

Reply:
Good idea, will generify this a bit more.

Reply:
The code can be used with anything already, by generifying the naming we basically get what we want already.